kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -882,29 +882,30 @@ protected void handleFindCoordinatorRequest(KafkaHeaderAndRequest findCoordinato
         String groupIdPath = GroupIdUtils.groupIdPathFormat(findCoordinator.getClientHost(),
                 findCoordinator.getHeader().clientId());
 
-        // Store group name to metadata store for current client.
+        // Store group name to metadata store for current client, use to collect consumer metrics.
         storeGroupId(groupId, groupIdPath)
-                .thenAccept(__ -> findBroker(TopicName.get(pulsarTopicName))
-                        .thenAccept(node -> {
-                            if (node.error() != Errors.NONE) {
-                                log.error("[{}] Request {}: Error while find coordinator.",
-                                        ctx.channel(), findCoordinator.getHeader());
-
-                                resultFuture.complete(KafkaResponseUtils
-                                        .newFindCoordinator(Errors.LEADER_NOT_AVAILABLE));
-                                return;
-                            }
+                .whenComplete((__, ex) -> {
+                    if (ex != null) {
+                        log.warn("Store groupId failed, the groupId might already stored.", ex);
+                    }
+                    findBroker(TopicName.get(pulsarTopicName))
+                            .thenAccept(node -> {
```

Comment:
what happens if findBroker fails ?
we are not going to complete `resultFuture` even in that case

Reply:
The `findBroker` method will never return a failed future, I think we don't need to handle it.

Reply:
"currently" it does not return a failure future, you we cannot know (especially for internal errors)
I believe it is better to add a handler for exceptions even in that case.
(just replace thenAccept with whenComplete)

Reply:
Yes, good suggestion. I have addressed.